### PR TITLE
Move arrow helpers from editor to tldraw

### PR DESCRIPTION
**Pull request by SomeHats**
With the new work on bindings, we no longer need to keep any arrows stuff hard-coded in `editor`, so let's move it to `tldraw` with the rest of the shapes.

Couple other changes as part of this:
- We had two different types of `WeakMap` backed cache, but we now only have one
- There's a new free-standing version of `createComputedCache` that doesn't need access to the editor/store in order to create the cache. instead, it returns a `{get(editor, id)}` object and instantiates the cache on a per-editor basis for each call.
- Fixed a bug in `createSelectedComputedCache` where the selector derivation would get re-created on every call to `get`

### Change Type

- [x] `sdk` — Changes the tldraw SDK
- [x] `improvement` — Improving existing features

#### Breaking changes
- `editor.getArrowInfo(shape)` has been replaced with `getArrowInfo(editor, shape)`
- `editor.getArrowsBoundTo(shape)` has been removed. Instead, use `editor.getBindingsToShape(shape, 'arrow')` and follow the `fromId` of each binding to the corresponding arrow shape
- These types have moved from `@tldraw/editor` to `tldraw`:
    - `TLArcInfo`
    - `TLArrowInfo`
    - `TLArrowPoint`
- `WeakMapCache` has been removed

### CODE REVIEW

**Comment by mimecuvalo** [comment]
200w

File: packages/store/src/lib/Store.ts [hunk 1]
```diff
@@ -989,3 +991,42 @@ class HistoryAccumulator<T extends UnknownRecord> {
...
+/**
+ * Free version of {@link Store.createComputedCache}.
+ *
+ * @example
+ * ```ts
+ * const myCache = createComputedCache('myCache', (editor: Editor, shape: TLShape) => {
+ *     return editor.getSomethingExpensive(shape)
+ * })
+ *
+ * myCache.get(editor, shape.id)
```
Comment (ds300) [inline packages/store/src/lib/Store.ts:1008]: cool stuff

File: packages/editor/src/lib/editor/Editor.ts [hunk 2]
```diff
@@ -4900,7 +4843,7 @@ export class Editor extends EventEmitter<TLEventMap> {
 		const parentId = typeof parent === 'string' ? parent : parent.id
 		const ids = this._parentIdsToChildIds.get()[parentId]
 		if (!ids) return EMPTY_ARRAY
-		return this._childIdsCache.get(ids, () => ids)
```
Comment (ds300) [inline packages/editor/src/lib/editor/Editor.ts:4903]: what was this even doing before 🤷🏼

File: packages/editor/api-report.md [hunk 3]
```diff
@@ -2096,57 +2075,6 @@ export type TLAnyBindingUtilConstructor = TLBindingUtilConstructor<any>;
...
-// @public (undocumented)
-export interface TLArrowBindings {
-    // (undocumented)
-    end: TLArrowBinding | undefined;
-    // (undocumented)
-    start: TLArrowBinding | undefined;
-}
-
-// @public (undocumented)
-export type TLArrowInfo = {
```
Comment (ds300) [inline packages/editor/api-report.md:2124]: I think these types were supposed to remain public in the `'tldraw'` package but it looks like they're not being exported from the index there?

**Review by ds300 — APPROVED** [review]
Looks good. Just noticed one issue with exports I think

File: packages/editor/api-report.md (hunk 3 above)
Comment (SomeHats) [reply packages/editor/api-report.md:2124]: ahh good catch, thank you!

**Pull request merged**